lib/Common/Memory/RecyclerTelemetryInfo.cpp

```diff
@@ -0,0 +1,256 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "CommonMemoryPch.h"
+
+
+#ifdef ENABLE_BASIC_TELEMETRY
+
+#include "Recycler.h"
+#include "HeapBucketStats.h"
+#include "BucketStatsReporter.h"
+#include <psapi.h>
+
+namespace Memory
+{
+#ifdef DBG
+#define AssertOnMainThread(recyclerTelemetryInfo, loc) { AssertMsg(recyclerTelemetryInfo->mainThreadID == GetCurrentThreadContextId(), STRINGIZE("Unexpected thread ID at " ## loc)); }
```

Comment:
I dont know if this works in the rental threading case (since a JsrtRuntime could get moved to being executed on another thread)- a quick glance in PageAllocator.h reveals it has support for updating the thread handle (UpdateThreadContextHandle) so you might need something like that. This would be for the case where folks are using JSRT with Chakra.dll and are using it in a rental-threaded way.

Reply:
Yes, I have a //TODO which is to go trough this and fix up my assertions on the current thread.  I initially had these just to confirm some of my assumptions.

Reply:
#ifdef'd these assertions out for now.  Will fixup asserts on a smaller subsequent PR.